### ISSUE TITLE
Scroll state not remember on home screen

### ISSUE THREAD

**Issue by owenlejeune** [Automattic/pocket-casts-android#1861]
### Description

If a user has enough subscriptions that they scroll off the screen, the scroll state is not remembered if the last podcast they viewed is below the fold.

### Step-by-step reproduction instructions

1. Make sure you have enough subscriptions that you can scroll on the home page
2. Scroll down below the fold
3. Tap to view a podcast that is below the fold
4. Return the home screen and scroll back to the top
5. Background and then foreground the app again
6. Observe that the scroll position will be reset from the top to the scroll position of the previously selected podcast.

### Did you search for existing bug reports?

- [X] I have searched for existing bug reports.

### Device, Operating system, and Pocket Casts app version

Google Pixel 7 w Android 14, Pocket Casts v7.57-rc-3 (9192)

**Comment by CookieyedCodes** [comment]
Agreed I would appreciate this, but if it is implemented I will need a button somewhere to jump to the top 😅, that dosent lag my phone with 1940+ feeds that may be difficult 😅 😉

**Comment by owenlejeune** [comment]
Ideally this shouldn't require a button like that. Currently, the behaviour is to jump through the list to the previously selected podcasts positions. This change would keep the list scrolled to wherever the user had scrolled rather than jumping through the list. You can see the existing behaviour here

https://github.com/Automattic/pocket-casts-android/assets/20568035/3e867c8d-f6d0-4404-bdcc-5c09b11701c0
